lib/grammers-client/src/client/auth.rs

```diff
@@ -240,7 +270,7 @@ impl Client {
         &mut self,
         token: &LoginToken,
         code: &str,
-    ) -> Result<tl::types::User, SignInError> {
+    ) -> Result<SignInResult, SignInError> {
```

Comment:
I am not too sure on this. I would argue requiring a password or signing up are "errors" because sign in itself failed. Forcing the user to handle the "error" case even when it returns `Ok` (as in, "hey, sign in failed, you actually need to sign up") doesn't feel right.

So I would argue sign up required / password needed should be part of sign in error.

Reply:
Okay, I agree about "sign up required", but I think "password needed" can be ok result. If we consider `signIn` as a separate function, then yes, it's an error, sign in failed. But if we look at the whole process, this is a fine result "hey, your token code is okay, but we need a password now".

However, I'm okay to move it to errors, this looks like a matter of taste after all. And telegram considers it as an error. Do you want to move it to the error?

Reply:
Yes, let's move it to errors, along with the sign up required.